### PR TITLE
0.9.x Build Troubleshooting Early December 2023 SGT

### LINKED ISSUE #812: Some macOS builds failing due to Python version conflicts
(Python 3.11 vs. Python 3.12.) Boost Python is apparently trying to use the wrong one, or something.

### PR DESCRIPTION
**Pull request by stephengtuggy**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [X] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation

Issues Fixed By This PR:
Fixes #812 for 0.9.x and following

Purpose:
- What is this pull request trying to do? Get Windows, Linux, and macOS CI builds all succeeding consistently
- What release is this for? 0.9.x
- Is there a project or milestone we should apply this to? 0.9.x

Issues Remaining, observed during testing:
#823 
#793

### CODE REVIEW

File: script/brew-install-for-cmake.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Vega Strike.  If not, see <https://www.gnu.org/licenses/>.
+
+set -e
+
+PACKAGE_SPEC="$1"
+LEAVE_LINKED="$2"
```
Comment (BenjamenMeyer) [inline script/brew-install-for-cmake.sh:30]: Would be good to have some comments describing how to use this script and the meaning of the parameters.

Comment (stephengtuggy) [reply script/brew-install-for-cmake.sh:30]: @BenjamenMeyer Agreed. It would be good. :-)

Comment (stephengtuggy) [reply script/brew-install-for-cmake.sh:30]: ... To finish the thought: Can we delay adding the comments until a future PR? Or do they need to be added right now, in your opinion?

Comment (BenjamenMeyer) [reply script/brew-install-for-cmake.sh:30]: We can delay

Comment (stephengtuggy) [reply script/brew-install-for-cmake.sh:30]: Okay, sounds good. I'll file a new issue for that documentation work

**Comment by BenjamenMeyer** [comment]
@stephengtuggy please rerun the Mac-CI builds so we can see what's broken.

**Comment by stephengtuggy** [comment]
> @stephengtuggy please rerun the Mac-CI builds so we can see what's broken.

@BenjamenMeyer I'd be happy to. Unfortunately, the Rerun button appears to be disabled. Maybe because the CI builds for this PR last ran so many months ago?

I plan to port some of the needed changes that were just merged into 0.8.x, into this PR branch. That should trigger a fresh run of the CI builds, and hopefully fix some or all of the remaining errors as well.

**Comment by BenjamenMeyer** [comment]
Perhaps just do a minor change (whitespace?) and push. Unfortunately it runs everything.

**Comment by stephengtuggy** [comment]
Closing in favor of #851 , which includes pretty much all of the important changes from this PR.

**Pull request closed without merging**